src/plugins/intel_gpu/src/graph/include/program_node.h

```diff
@@ -125,21 +125,13 @@ struct program_node {
     void set_preferred_impl_type(impl_types impl) { impl_type = impl; }
     impl_types get_preferred_impl_type() const { return impl_type; }
 
-    std::vector<program_node*> const& get_dependencies() const { return dependencies; }
-    std::vector<std::pair<program_node*, int>> const& get_dependencies_new() const { return dependencies_new; }
-    program_node& get_dependency(size_t idx) const { return *dependencies.at(idx); }
-    std::pair<program_node*, int32_t> get_dependency_new(size_t idx) const { return dependencies_new.at(idx); }
+    std::vector<std::pair<program_node*, int32_t>> const& get_dependencies() const { return dependencies; }
+    std::pair<program_node*, int32_t> get_dependency(size_t idx) const { return dependencies.at(idx); }
```

Comment:
Looks like in most of the cases we need program_node ptr and port id is not needed. So maybe method can return ptr only, so less changes (and less future code) will be needed? We can have a separate method which return pair, and it may be used if port id is really needed.